components/automate-ui/src/app/page-components/simple-line-graph/simple-line-graph.component.ts

```diff
@@ -0,0 +1,342 @@
+  }
+
+  private renderTooltips(): void {
+    // these numbers are specific to its container
+    const localWidth = this.width - this.margin.right - this.margin.left - 34;
+    const thisRange = [localWidth, -this.margin.right];
+    const thisScale = d3.scaleLinear()
+      .domain(this.domainX)
+      .range(thisRange);
+
+    const tooltips = this.containerSelection.selectAll('div.graph-tooltip')
+      .data(this.data, d => d.daysAgo);
+    tooltips.exit().remove();
+    tooltips.enter().append('div')
+      .attr('class', (_d, i: number) => `graph-tooltip elem-${i}`)
+      .merge(tooltips)
+      .text(d => `Checked in ${Math.round(d.percentage)}%`)
+      .style('left', (d, i: number) => {
+        const left = thisScale(d.daysAgo);
+        if ( i === 0 ) { return `${localWidth + this.margin.right}px`; }
+        return `${left}px`;
+      })
+      .style('top', d => `${this.yScale(d.percentage) + this.margin.top}px`);
+  }
+
+  private renderLabelButtons(): void {
+    // these numbers are specific to its container
+    const thisRange = [this.width - 48, 53];
+    const thisScale = d3.scaleLinear()
+      .domain(this.domainX)
+      .range(thisRange);
+
+    const labels = this.containerSelection.selectAll('.graph-button')
+      .data(this.data, d => d.daysAgo);
+    labels.exit().remove();
+    labels.enter().append('button')
+      .call(parent => {
+        parent.append('div')
+          .attr('class', (_d, i: number) => `inner elem-${i}`);
+      })
+      .attr('class', (_d, i: number) => `graph-button elem-${i}`)
+      .merge(labels)
+      .transition().duration(0)
+        .style('bottom', `${this.margin.bottom / 2}px`)
+      .transition().duration(1000)
+        .style('bottom', `${this.margin.bottom / 2}px`)
+        .style('left', d => {
+          if ( this.xData.length > 7 ) {
+            return `${this.xScale(d.daysAgo) - 30}px`;
+          } else {
+            return `${thisScale(d.daysAgo) - 30}px`;
+          }
+        })
+      .call(parent => {
+        parent.select('.inner')
+          .text(p => this.formatLabels(p.daysAgo));
+      });
+
+      // add all listeners
+    this.containerSelection.selectAll('.graph-button')
+      // add class to rotate labels when more than comfortable to fit in space
+      .classed('turnt', () => this.xData.length > 7)
+
+      .on('mouseenter', () => {
+        this.handleHover(d3.event);
+      })
+      .on('mouseout', () => {
+        this.AllDeactivate();
+      })
+      // focus styles
+      .on('focus', () => {
+        this.handleHover(d3.event);
+      })
+      .on('focusout', () => {
+        this.AllDeactivate();
+      })
+      .on('click', () => {
+        this.handleClick(d3.event);
+      });
+  }
+
+  private formatLabels(daysAgo: number): string {
+    switch (daysAgo) {
+      case 0:
+        return '24 hrs ago';
+        break;
+      default:
+        return `${daysAgo + 1} days ago`;
+    }
+  }
+
+  private renderGrid() {
```

Comment:
All through rendering the grid is where you'll see the margin variables come into play.  This is standard d3.